src/gretel_synthetics/generator.py

```diff
@@ -0,0 +1,236 @@
+from typing import TYPE_CHECKING, Callable, List, Iterable, Optional, Tuple
+from dataclasses import dataclass, asdict
+from collections import namedtuple
+
+import cloudpickle
+
+import sentencepiece as spm
+import tensorflow as tf
+
+from .model import _build_sequential_model
+
+if TYPE_CHECKING:
+    from .config import BaseConfig, LocalConfig
+
+
+def _load_tokenizer(store: "LocalConfig") -> spm.SentencePieceProcessor:
+    sp = spm.SentencePieceProcessor()
+    sp.Load(store.tokenizer_model)
+    return sp
+
+
+def _prepare_model(
+    sp: spm.SentencePieceProcessor, batch_size: int, store: "LocalConfig"
+) -> tf.keras.Sequential:  # pragma: no cover
+    model = _build_sequential_model(
+        vocab_size=len(sp), batch_size=batch_size, store=store
+    )
+
+    load_dir = store.checkpoint_dir
+
+    model.load_weights(tf.train.latest_checkpoint(load_dir)).expect_partial()
+
+    model.build(tf.TensorShape([1, None]))
+    model.summary()
+
+    return model
+
+
+def _load_model(
+    store: "LocalConfig",
```

Comment:
Can potentially avoid the `LocalConfig` forward ref here based on import logic change